lib/src/packages_file.dart

```diff
@@ -100,17 +106,24 @@ PackageConfig parse(
           "Package URI as location for package", source, separatorIndex + 1));
       continue;
     }
-    if (!packageLocation.path.endsWith('/')) {
-      packageLocation =
-          packageLocation.replace(path: packageLocation.path + "/");
+    var path = packageLocation.path;
+    if (!path.endsWith('/')) {
+      path += "/";
+      packageLocation = packageLocation.replace(path: path);
     }
     if (packageNames.contains(packageName)) {
       onError(PackageConfigFormatException(
           "Same package name occured more than once", source, start));
       continue;
     }
+    var rootUri = packageLocation;
+    if (path.endsWith("/lib/")) {
```

Comment:
Will this always be relative to something else?  That is, can `path` ever be `lib/`?

Reply:
The "root" package will have a path of `lib/`.

Reply:
Good point, I should resolve the URI against the base URI before doing this check.